pkg/apiserver/watch.go

```diff
@@ -56,7 +56,7 @@ func (h *WatchHandler) setSelfLinkAddName(obj runtime.Object, req *http.Request)
 	return h.linker.SetSelfLink(obj, newURL.String())
 }
 
-func getWatchParams(query url.Values) (label, field labels.Selector, resourceVersion string, err error) {
+func getWatchParams(query url.Values, version, apiResource string) (label, field labels.Selector, resourceVersion string, err error) {
```

Comment:
Optional suggestion: list does this exact same thing; perhaps you could factor that into a common function?